spec/models/course_spec.rb

```diff
@@ -846,4 +847,48 @@
       expect(course.pages_edited).not_to include(article2)
     end
   end
+
+  describe '#tracked_revisions' do
+    let(:course) { create(:course, start: 1.year.ago) }
+    let(:user) { create(:user, id: 3, username: 'Amit') }
+    let(:article) { create(:article, namespace: 0, deleted: false) }
+    let(:article2) { create(:article, namespace: 0, deleted: false) }
+
+    before do
+      create(:revision, date: course.start + 1.minute,
```

Comment:
I think, rather than creating the revisions explicitly, this test should choose a user and course date range such that the update will create those articles and revisions and we're relying completely on the update flow to generate the article, revision and course stats data.